include/Factory.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "King.h"
+#include "Queen.h"
+#include "Rook.h"
+#include "Bishop.h"
+#include "Knight.h"
+#include "Pawn.h"
+#include "macros.h"
+
+#include <functional>
+#include <memory>
+#include <map>

```

Comment:
Include Organize the includes in a more structured way. Place the system headers first, followed by your project headers, and use separate sections with comments to distinguish between them.